gitpod-io/gitpod
# [loadgen] fix application of runtime stress & add network stress

## Description

* Remove PVC loadgen configurations
* Add network stress & enable network limiting
* Add adding fields to result to help with inspection
* Apply pressure at runtime, `stress-ng` was missing as it's not part of the underlying image

- [x] tune network parameters, the load averages are very high :upside_down_face: , assert network limiting is enabling

## Related Issue(s)

Fixes # n/a

## How to test

1. Observe `stats.json` within the `results` directory of where you started `loadgen`
2. Find a workspace where the CloneURL contains `workspace-stress`, and observe it has WorkspaceName and WorkspaceImage fields, copy the InstanceID
3. Enter the pod associated with the InstanceID, and observe that `ps axf` shows each stress test (there are five, four require stress-ng)
4. Watch the # of connections from within a `workspace-stress` pod via `watch "ss | wc"`
   * @Furisto wdyt? Would it be more appropriate to do `ss -t`?

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
